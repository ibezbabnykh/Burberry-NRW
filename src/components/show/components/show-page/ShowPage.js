import React, { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';

import { useShow } from 'hooks';
import { loadShowDetails, loadEpisodeList } from '../../store/actions';
import { Spinner, ErrorMessage } from 'components/common';
import ShowDetails from '../show-details';
import EpisodeList from '../episode-list';

const renderDetails = (error, loading, detailsData, episodeListData) => {
  if (loading) {
    return <Spinner />;
  }

  if (error) {
    return <ErrorMessage text={error} />;
  }

  return (
    <div className="show-page">
      <ShowDetails data={detailsData} />
      <EpisodeList data={episodeListData} />
    </div>
  );
};

const ShowPage = () => {
  const { id } = useParams();
  const { detailsData, episodeListData, error, loading } = useShow();
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(loadShowDetails(id));
    dispatch(loadEpisodeList(id));
  }, [id, dispatch]);

  return (
    renderDetails(error, loading, detailsData, episodeListData)
  );
}

export default ShowPage;