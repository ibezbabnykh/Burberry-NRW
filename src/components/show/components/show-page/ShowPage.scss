@import '../../../../assets/style/variables.scss';

.show-page {
    
}