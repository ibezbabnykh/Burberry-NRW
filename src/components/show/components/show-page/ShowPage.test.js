import { Provider } from 'react-redux';
import ShowPage from './ShowPage';

const mockStore = configureStore();

jest.mock('react-router-dom', () => ({
  useParams: () => ({ id: 2 }),
}));

const mockShowDetails = {
    image: {
      medium: 'https://via.placeholder.com/250x140',
      original: 'https://via.placeholder.com/1280x720'
    },
    name: 'show name',
    summary: 'show summary'
};

const mockEpisodeLists = [
  {
    season: '1',
    values: [
      { id: 1, number: 1, season: 1, name: 'episode 1x1', airdate: '2016-04-11' },
      { id: 2, number: 2, season: 1, name: 'episode 1x2', airdate: '2016-04-15' }
    ]
  },
  {
    season: '2',
    values: [
      { id: 3, number: 1, season: 2, name: 'episode 2x1', airdate: '2017-04-11' },
      { id: 4, number: 2, season: 2, name: 'episode 2x2', airdate: '2017-04-15' }
    ]
  }
];

describe('ShowPage component', () => {
  let store;
  let component;

  const initialState = {
    show: {
      details: mockShowDetails,
      episodeList: mockEpisodeLists
    }
  };

  beforeEach(() => {
    store = mockStore(initialState);

    store.dispatch = jest.fn();
 
    component = renderer.create(
      <Provider store={store}>
        <ShowPage />
      </Provider>
    );
  });

  it('should render', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});