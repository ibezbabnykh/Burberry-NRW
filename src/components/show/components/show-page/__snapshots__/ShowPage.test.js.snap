// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowPage component should render 1`] = `
<div
  className="loadingio-spinner-rolling"
>
  <div
    className="spinner"
  >
    <div />
  </div>
</div>
`;
